src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -615,5 +598,51 @@ private void CheckRefReadOnlySymbols(MethodSymbol symbol)
                 _factory.CompilationState.ModuleBuilderOpt?.EnsureIsReadOnlyAttributeExists();
             }
         }
+
+#if DEBUG
+        private class LocalRewritingValidator : BoundTreeWalker
+        {
+            /// <summary>
+            /// Asserts that no unexpected nodes survived local rewriting.
+            /// </summary>
+            [Conditional("DEBUG")]
+            public static void Validate(BoundNode node)
+            {
+                new LocalRewritingValidator().Visit(node);
+            }
+
+            public override BoundNode Visit(BoundNode node)
+            {
+                if (!(node is BoundExpression))
```

Comment:
Should this walk into lambda and local function bodies as well?

Reply:
Good catch. Thanks

---
In reply to: [161901756](https://github.com/dotnet/roslyn/pull/23961#discussion_r161901756) [](ancestors = 161901756)